### PR TITLE
Add exercise and user exercise model and exercise view page for patients

### PR DESCRIPTION
**Pull request by asadeg02**
2 new models "Exercise" and "UserExercise" are added with reference to #14 and #15.
"userExercise" model models the association between each patient and their exercises. In this version both practice and reference exercises are modeled in one single model as "user Exercise" which might be subject to change after implementing our first working version using anchor.

First view page for the patients displaying their exercises and their latest session date is also added with reference to #16.

userExercise model has made use of 1 important assumption:
1- every patient can only have one reference exercise for each exercise type. (cilician might update the reference exercise but the triple (userId, exerciseId, Reference) always defines a unique document in "userExercise" collection.
so any change in business logic of application violating this assumption, might introduce a change to our current logic for modeling the association between users ans their exercises .

### CODE REVIEW

File: server/api/userExercise.js [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+    method: 'POST',
+    path: '/userexercise/dummy/reference',
+    config: {
+      auth: {
+        strategies: ['simple', 'jwt', 'session']
+      },
+      validate: {
+        payload: UserExercise.refPayload
+      }
+    },
```
Comment (shreyapandit) [inline server/api/userExercise.js]: Why is this extra route needed? Why not instead pass empty body frames to the original route? Same for practice below

File: server/models/refexercise.js [hunk 2]
```diff
@@ -28,18 +28,30 @@ RefExercise.collection = 'refexercises';
 
 RefExercise.schema = Joi.object().keys({
   _id: Joi.object(),
+  userId: Joi.string().required(),
+  exerciseId: Joi.string().required(),
+  //we can also define a boolean field as isRefference instead 
+  type: Joi.string().valid('Reference','Practice').required(),
+  refferenceId: Joi.string().required(),
+  isActive: Joi.boolean().required(),
```
Comment (shreyapandit) [inline server/models/refexercise.js]: We should merge this with the userexercise model

File: server/web/templates/userexercise/viewexercises.handlebars [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                    <label for="mySearch">Search for exercises</label>
+                    <input type="search" id="search" name="q" placeholder="search for an exercise" onkeyup="f()">
+                    <button onclick="search()">Search</button>
+                    <span class="validity"></span>
+                 </div>
+               </form>
+               <br>    
+               <!div class="card">
+                    <!div style="padding: 20px">
+                       <ul id ="ul" class = "ul">
```
Comment (shreyapandit) [inline server/web/templates/userexercise/viewexercises.handlebars]: Can we remove the inline styles and use preavailable css classes/refactor them into styles.css as a reusable class?

File: server/web/templates/userexercise/createref.handlebars [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<div class="container-fluid h-100">
+  <div class="row h-100">
+    {{>dashboardNav}}
+    <div class="col-sm-9 col-md-10">
+      <div style="padding: 30px">
+        <h3>Create Reference Exercise</h3>
+        <hr>
+        <div class="card">
+          <div style="padding: 20px">
+            <form id="form">
```
Comment (shreyapandit) [inline server/web/templates/userexercise/createref.handlebars]: All this css will sprawl out of hand if we keep doing this across templates. Lets try to refactor/reuse as much as we can and condense these classes into a single file

File: server/web/templates/userexercise/createpractice.handlebars [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      <div style="padding: 30px">
+        <h3>Create practice Exercise</h3>
+        <hr>
+        <div class="card">
+          <div style="padding: 20px">
+            <form id="form">
+              <div id="formFields"></div>
+              <div class="form-group">
+                <hr>
+                <label for="userId">Select the user:</label>
```
Comment (shreyapandit) [inline server/web/templates/userexercise/createpractice.handlebars]: Again, to fix css

File: server/web/templates/userexercise/beginsession.handlebars [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+                                height: 800px;
+                                width: 1100px;"></canvas>
+
+        <h1 id="line1" style="position: absolute; right: 0; top: 300px">To begin the session,</h1>
+        <h1 id="line2" style="position: absolute; right: 100px; top: 350px"> click start.<h1>
+        <h3 id="line3" style="position: absolute; right: 100px; top: 350px color: blue; visibility: hidden;"> What would you like to do?<h3>
+        <h4 style="position: absolute; right: 45px; top: 400px; color: blue;">Current Exercise: '{{exercise.exerciseName}}'<h4>
+        <div class="btn btn-primary" id="startSession" onclick="start()" style="position: absolute; 
+                                                       right: 50px; 
+                                                       top: 450px;
```
Comment (shreyapandit) [inline server/web/templates/userexercise/beginsession.handlebars]: Why is this whole block of css here? This is a code smell and should be removed

File: server/web/public/scripts/userexercise/createpractice.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ 
+
+$('#create').click((event) => {
+  event.preventDefault();
+  const values = {};
+  values.userId = $('#userId').val();
+  values.exerciseId = $('#exerciseId').val();
+  $.ajax({
+    type: 'POST',
+    url: '/api/userexercise/dummy/practice',
```
Comment (shreyapandit) [inline server/web/public/scripts/userexercise/createpractice.js]: I think you can get rid of the dummy route and just dummy data/empty [] for the kinect data as a placeholder.  No need to create dummy routes specially for these?

File: server/web/templates/userexercise/viewexercises.handlebars (hunk 3 above)
Comment (shreyapandit) [reply server/web/templates/userexercise/viewexercises.handlebars]: This is a syntax error ```!div class="card"``` Please fix this

**Comment by shreyapandit** [comment]
Please merge with master, the branch seems to be behind

File: server/web/templates/clinician/viewpatientexercises.handlebars [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      {{>dashboardNav}}
+        <div class="col-sm-9 col-md-10">
+            <div style="padding: 30px; color: #525B74;">
+                <br><br>
+                <h1 style= "text-align: center; font-size: 50px; font-weight: bold;" >Select an exercise.</h1>
+                <h3 align="center" style="color:#428BCA; font-size: 20px;">You are currently working with patient {{patientId}}.</h3>
+                <br>
+                <form style= "padding-left: 50px;">
+                  <!div style="padding: 0px;">
+                    <label for="mySearch"></label>
```
Comment (shreyapandit) [inline server/web/templates/clinician/viewpatientexercises.handlebars]: Syntax error?

Comment (asadeg02) [reply server/web/templates/clinician/viewpatientexercises.handlebars]: I have done all the code clean up for clinician pages on its own branch, I have made other changes to clinician pages as well!so this error won't exist after merging all the branches.

**Pull request merged**